client/network/src/transport.rs

```diff
@@ -121,11 +137,23 @@ pub fn build_transport(
 		core::upgrade::SelectUpgrade::new(yamux_config, mplex_config)
 	};
 
-	let transport = transport
+	let tcp_transport = transport
 		.upgrade(upgrade::Version::V1Lazy)
 		.authenticate(authentication_config)
 		.multiplex(multiplexing_config)
-		.timeout(Duration::from_secs(20))
+		.timeout(Duration::from_secs(20));
+
+	let quic_transport = if enable_quic {
+		OptionalTransport::some(build_quic_transport(&keypair))
+	} else {
+		OptionalTransport::none()
+	};
+
+	let transport = OrTransport::new(quic_transport, tcp_transport)
+		.map(|either_output, _| match either_output {
+			EitherOutput::First((peer_id, muxer)) => (peer_id, StreamMuxerBox::new(muxer)),
+			EitherOutput::Second((peer_id, muxer)) => (peer_id, StreamMuxerBox::new(muxer)),
+		})
```

Comment:
I remember a similar construct in @melekes's PR for WebRTC, but in the end it worked out without it. Was the trick to use `tcp_transport.or_transport(quick_transport)` instead of `OrTransport::new`?

Reply:
I'm using the same construct. I don't think there is a way to avoid it.